pkg/controllers/landscaperdeployments/reconcile.go

```diff
@@ -97,13 +97,10 @@ func (c *Controller) mutateInstance(ctx context.Context, log logr.Logger, deploy
 		instance.Spec.ServiceTargetConfigRef.Namespace = serviceTargetConf.GetNamespace()
 	}
 
-	// The instance has a generated name and therefore may be not set yet.
-	// The hash is therefore calculated with the name of the deployment.
-	// Since the name of the instance is derived from the deployment name, which is unique in a namespace,
-	// the hash value is also unique.
-	h := sha1.New()
-	id := h.Sum([]byte(deployment.Name))
-	instance.Spec.ID = hex.EncodeToString(id[:4])
+	if len(instance.Spec.ID) == 0 {
+		id := guuid.New()
+		instance.Spec.ID = id.String()[:8]
```

Comment:
Not sure how probable clashes are with approach? What about checking if this ID is already in use?

Reply:
That would be nice. However we would need a solution that survives restarts of the controller.
Is there anything that you have in mind?

Reply:
I fear I do not understand your reply :-) I only recommend to reject and generate a new guid if the first one is already in use by an already existing instance.